sdk/core/core-amqp/src/auth/iotSas.ts

```diff
@@ -1,23 +1,23 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-import { SasTokenProvider } from "./sas";
-import { TokenInfo } from "./token";
+import { SharedKeyCredential } from "./sas";
+import { AccessToken } from "@azure/core-http";
 import { Buffer } from "buffer";
 
 /**
  * @class IotSasTokenProvider
  * @ignore
  * Defines the SasTokenProvider for IotHub.
  */
-export class IotSasTokenProvider extends SasTokenProvider {
+export class IotSasTokenProvider extends SharedKeyCredential {
```

Comment:
Do we know where this gets used? If we don't have a use case for it yet, then lets remove this from `core-amqp`. We can bring it back when we have some feedback for the need for it

Reply:
Got it, then lets just rename it to use "SharedKeyCredential" instead of "SasTokenProvider"